### PR TITLE
DM-29421: Add AP plots in analysis_ap using analysis_tools

### PR DESCRIPTION
**Pull request by mrawls**
This PR adds a new plotting capability for DiaSources on the sky. The simple version of this plot is here in `analysis_tools`, and a specialized version for a specific AP QA data set is in `analysis_ap`.

### CODE REVIEW

**Comment by mrawls** [comment]
As part of the review, I need some assistance getting the modified pipeline to run before this is ready to merge. At the moment, when I run
```
pipetask run -b /sdf/data/rubin/u/parejko/cosmos-DM-36199/repo/butler.yaml -i ap_verify-output -o test_metric_plot -p $ANALYSIS_TOOLS_DIR/pipelines/apCcdVisitQualityCore.yaml
```

The result is as follows
```
lsst.daf.butler.cli.utils ERROR: Caught an exception, details are in traceback:
Traceback (most recent call last):
  File "/sdf/group/rubin/sw/conda/envs/lsst-scipipe-4.1.0/share/eups/Linux64/ctrl_mpexec/g14023acf34+0a8cb3cd94/python/lsst/ctrl/mpexec/cli/cmd/commands.py", line 160, in run
    if (qgraph := script.qgraph(pipelineObj=pipeline, **kwargs, show=show)) is None:
  File "/sdf/group/rubin/sw/conda/envs/lsst-scipipe-4.1.0/share/eups/Linux64/ctrl_mpexec/g14023acf34+0a8cb3cd94/python/lsst/ctrl/mpexec/cli/script/qgraph.py", line 196, in qgraph
    qgraph = f.makeGraph(pipelineObj, args)
  File "/sdf/group/rubin/sw/conda/envs/lsst-scipipe-4.1.0/share/eups/Linux64/ctrl_mpexec/g14023acf34+0a8cb3cd94/python/lsst/ctrl/mpexec/cmdLineFwk.py", line 579, in makeGraph
    qgraph = graphBuilder.makeGraph(
  File "/sdf/group/rubin/sw/conda/envs/lsst-scipipe-4.1.0/share/eups/Linux64/pipe_base/ge08f41d946+f12b9a963a/python/lsst/pipe/base/graphBuilder.py", line 1335, in makeGraph
    scaffolding = _PipelineScaffolding(pipeline, registry=self.registry)
  File "/sdf/group/rubin/sw/conda/envs/lsst-scipipe-4.1.0/share/eups/Linux64/pipe_base/ge08f41d946+f12b9a963a/python/lsst/pipe/base/graphBuilder.py", line 568, in __init__
    datasetTypes = PipelineDatasetTypes.fromPipeline(pipeline, registry=registry)
  File "/sdf/group/rubin/sw/conda/envs/lsst-scipipe-4.1.0/share/eups/Linux64/pipe_base/ge08f41d946+f12b9a963a/python/lsst/pipe/base/pipeline.py", line 1143, in fromPipeline
    pipeline = list(pipeline)
  File "/sdf/group/rubin/sw/conda/envs/lsst-scipipe-4.1.0/share/eups/Linux64/pipe_base/ge08f41d946+f12b9a963a/python/lsst/pipe/base/pipeline.py", line 665, in toExpandedPipeline
    taskDefs.append(self._buildTaskDef(label))
  File "/sdf/group/rubin/sw/conda/envs/lsst-scipipe-4.1.0/share/eups/Linux64/pipe_base/ge08f41d946+f12b9a963a/python/lsst/pipe/base/pipeline.py", line 708, in _buildTaskDef
    overrides.applyTo(config)
  File "/sdf/group/rubin/sw/conda/envs/lsst-scipipe-4.1.0/share/eups/Linux64/pipe_base/ge08f41d946+f12b9a963a/python/lsst/pipe/base/configOverrides.py", line 285, in applyTo
    setattr(tmpConfig, finalField, value)
  File "/sdf/group/rubin/sw/conda/envs/lsst-scipipe-4.1.0/share/eups/Linux64/pipe_tasks/ge54c72f270+dce6a40c95/python/lsst/pipe/tasks/configurableActions/_configurableActionStructField.py", line 196, in __setattr__
    valueInst = value(__name=name, __at=at, __label=label)
TypeError: 'str' object is not callable
```

**Comment by mrawls** [comment]
This should be all set now and ready for your actual review!

File: python/lsst/analysis/tools/actions/plot/diaSkyPlot.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+    )
+    dec = Field[str](
+        doc="Name of Dec column",
+        optional=False,
+    )
+
+
+class DiaSkyPlot(PlotAction):
+    """Generic pseudo base class for plotting DiaSources
+    (or DiaObjects) on the sky.
```
Comment (natelust) [inline python/lsst/analysis/tools/actions/plot/diaSkyPlot.py:75]: This doc string seems awkward for some reason, is the line break in the right place?

**Review by natelust — APPROVED** [review]
Some comments to consider

**Pull request merged**